### PR TITLE
cleanup(create-space-dialog): remove template selection code

### PR DESCRIPTION
**Pull request by rohitkrai03**
- When creating a Space the first choice is to set the Development Process. This should be defaulted to Agile and the field should be hidden.

Screenshots - 

screenshot 2018-10-12 at 12 49 37 am

## Updates - 
- Now the backend defaults to Agile template so there is no need to have any code related to template selection.
- This PR removes all the template selection code and updates tests.

New Screenshot - 
screenshot 2018-10-23 at 10 37 22 pm

cc: @fabric8-ui/uxd

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @rohitkrai03 for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin.

File: src/app/space/add-space-overlay/add-space-overlay.component.ts [hunk 1]
```diff
@@ -60,7 +60,9 @@ export class AddSpaceOverlayComponent implements OnInit {
     }));
     this.subscriptions.push(this.spaceTemplateService.getSpaceTemplates()
       .subscribe((templates: ProcessTemplate[]) => {
-        this.spaceTemplates = templates.filter(t => t.attributes['can-construct']);
+        this.spaceTemplates = templates.filter(t => {
+          return t.attributes['can-construct'] && (t.attributes.name === 'Agile' || t.id === 'f405fa41-a8bb-46db-8800-2dbe13da1418');
```
Comment (joshuawilson) [inline src/app/space/add-space-overlay/add-space-overlay.component.ts]: This ID feels like a risk. What happens if the ID changes on the backend.

Comment (joshuawilson) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: I can accept that this is the primary key and shouldn't change, it still just feels odd having a PK in the UI.

Comment (joshuawilson) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: that should really be abstracted away

Comment (rohitkrai03) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: Do you have any other suggestions on how we could filter out just `Agile` template?

Comment (joshuawilson) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: no, maybe @kwk can answer that

Comment (kwk) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: If you don't send a template at all we will default for SDD on the backend: https://github.com/fabric8-services/fabric8-wit/blob/8674b7b56a9226b3bbd6ebf7b0179f127ff6a0a2/controller/space.go#L96 . Maybe it is time to change this to the Agile template and then you don't have to send anything?

Comment (rohitkrai03) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: @kwk That would be a better solution in my opinion. What do you think @joshuawilson?

Comment (joshuawilson) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: yes, changing the backend to default to Agile is much cleaner

Comment (kwk) [reply src/app/space/add-space-overlay/add-space-overlay.component.ts]: @joshuawilson I've create a task here https://github.com/fabric8-services/fabric8-wit/issues/2317

**Comment by joshuawilson** [comment]
I'm undesided if we should merge this and fix it later or wait. I'm inclined to wait, unless the backend change is going to take long.

**Comment by joshuawilson** [comment]
I've changed it to WIP since we are waiting for the backend change.

**Comment by jarifibrahim** [comment]
I have raised a PR on the backend https://github.com/fabric8-services/fabric8-wit/pull/2323

**Comment by joshuawilson** [comment]
Thanks @jarifibrahim, sounds great.

**Comment by jarifibrahim** [comment]
@rohitkrai03 @joshuawilson The e2e-tests rely on space creation wizard. Removing the template selection option might break some tests.

**Comment by rohitkrai03** [comment]
@jarifibrahim Thanks for the update. We'll have to inform QE team for this change so that the e2e tests can be updated.

**Comment by rohitkrai03** [comment]
@joshuawilson I've updated this PR to remove any code related to template selection. @ppitonak This may break some e2e tests because the UI for template selection is removed now.

**Comment by joshuawilson** [comment]
@fabric8-ui/uxd @fabric8-ui/docs @fabric8-ui/test @rkratky please review

**Comment by ljelinkova** [comment]
@rohitkrai03 @joshuawilson Thanks for the heads up. I've checked the e2e tests and it will work with this change.

**Comment by joshuawilson** [comment]
> @rohitkrai03 @joshuawilson Thanks for the heads up. I've checked the e2e tests and it will work with this change.

The tests don't have to set the Development Process? Does it just use the defaults now?

**Comment by joshuawilson** [comment]
@rohitkrai03 please submit this as a new PR since we had to revert from before it.

**Comment by Preeticp** [comment]
Docs have been updated via https://github.com/fabric8io/fabric8-online-docs/pull/447

**Pull request merged**